### ISSUE TITLE
Variable Expansion in vars is not working as expected

### ISSUE THREAD

**Issue by gregorydlogan** [robertdebock/ansible-role-bootstrap#24]
**Describe the bug**
Neither ansible_user, nor bootstrap_user are being honoured at runtime.

**To Reproduce**
1. I'm using a vanilla CentOS 7 LXC instance, but I imagine this would happen regardless.  Host is called cent7ct in this example
2. Ensure you can log in manually as root@cent7ct, accept hostkey, etc
3. Run the following playbook
```
---

- hosts: lxc,vm
  gather_facts: no
  user: root

  roles:
    - robertdebock.bootstrap
  vars:
    bootstrap_user: root
    bootstrap_remote_user: root
    bootstrap_retries: 0
```

**Expected behavior**
A successful install of the various bootstrap bits

**Error**
```
TASK [robertdebock.bootstrap : register apk] **********************************************************************************************************************************************************************
task path: /home/greg/.ansible/roles/robertdebock.bootstrap/tasks/main.yml:27
<cent7ct> ESTABLISH SSH CONNECTION FOR USER: None
<cent7ct> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/greg/.ansible/cp/f513c56a8b -tt cent7ct 'apk --version'
<cent7ct> (255, '', "Warning: Permanently added 'cent7ct,192.168.0.245' (ECDSA) to the list of known hosts.\r\nPermission denied (publickey,gssapi-keyex,gssapi-with-mic,password).\r\n")
fatal: [cent7ct]: UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: Warning: Permanently added 'cent7ct,192.168.0.245' (ECDSA) to the list of known hosts.\r\nPermission denied (publickey,gssapi-keyex,gssapi-with-mic,password).", 
    "unreachable": true
}
```

Note the line `ESTABLISH SSH CONNECTION FOR USER: None`

**Environment**
- Control node OS: Debian 9, up to date as of 2019-02-21
- Control node Ansible version: 2.7.7 from pip
- Managed node OS: CentOS 7 from a while ago (dated 20171212)

**Additional context**
Patching https://github.com/robertdebock/ansible-role-bootstrap/commit/4d0456093d601b4e5d9b80fa2bca0ef391678ff3#diff-6d62b73b3f46cfa11f6466d08cf45303R4 to specify root on this line works as expected, so I'm guessing that the variable resolution is occurring earlier than expected.

**Comment by jkirk** [comment]
@gregorydlogan Did you try this?
```yaml
---

- hosts: lxc,vm
  gather_facts: no
  user: root

  roles:
    - role: robertdebock.bootstrap
      bootstrap_remote_user: root
```

I've learned the hard way that the `vars` keyword changes the behavior on when / how the variables are set. I haven't found anything in docs explaining the difference but omitting the `vars` keywords seems to have a "stronger" effect on the variable.

**But** still, something feels wrong. My setup looks like this.
I usually have something like this in my `.ssh/config` file:
```
Host *.example.com
  User sysadmin                  
```
(This is, because after the bootstrap procedure, I will use the user `sysadmin` to do all the work.)

I think `bootstrap.yml` should look like this:
```
- name: Prepare system to be managed by Ansible
  hosts: all
  become: false
  gather_facts: false
  roles:
    - role: robertdebock.bootstrap
#     bootstrap_remote_user: root
#   - { role: user, users: [ 'sysadmin ], groupname: 'sysadmin', admin: True }
```
`bootstrap_user` defaults to `root`, so no need to set it here.
 (Please also explain the difference between the variables `bootstrap_user` and `bootstrap_remote_user`. I looks like that only `bootstrap_user` should be set.)

And this is what happens when running without `-u root`:
```
% ansible-playbook -D --limit myserver.example.com bootstrap.yml

PLAY [Prepare system to be managed by Ansible] *************************************************************
[...]
TASK [robertdebock.bootstrap : test connection] ************************************************************
fatal: [myserver.example.com]: FAILED! => {"changed": false, "elapsed": 4, "msg": "timed out waiting for ping module test success: SSH Error: data could not be sent to remote host \"myserver.example.com\". Make sure this host can be reached over ssh"}
...ignoring

TASK [robertdebock.bootstrap : register apk] ***************************************************************
ok: [myserver.example.com]
[...]

TASK [robertdebock.bootstrap : gather facts] ***************************************************************
fatal: [myserver.example.com]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to remote host \"myserver.example.com\". Make sure this host can be reached over ssh", "unreachable": true}
        to retry, use: --limit @/home/jkirk/projects/solsystem-ansible/bootstrap.retry

PLAY RECAP *************************************************************************************************
myserver.example.com         : ok=11   changed=0    unreachable=1    failed=0
```

And this is what happens when running with `-u root`:
```
ansible-playbook -vvv -D -u root --limit myserver.example.com bootstrap.yml
[...]
TASK [robertdebock.bootstrap : test connection] 
task path: /home/jkirk/projects/solsystem-ansible/roles-generic/robertdebock.bootstrap/tasks/main.yml:14
wait_for_connection: attempting ping module test
Using module file /home/jkirk/software/venv/local/lib/python2.7/site-packages/ansible/modules/system/ping.py
<myserver.example.com> ESTABLISH SSH CONNECTION FOR USER: root
<myserver.example.com> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/jkirk/.ansible/cp/ee6f662bfd myserver.example.com '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<myserver.example.com> (0, '\n{"invocation": {"module_args": {"data": "pong"}}, "ping": "pong"}\n', '')
ok: [myserver.example.com] => {
    "changed": false, 
    "elapsed": 0
}
[...]
TASK [robertdebock.bootstrap : register apk] 
task path: /home/jkirk/projects/solsystem-ansible/roles-generic/robertdebock.bootstrap/tasks/main.yml:27
<myserver.example.com> ESTABLISH SSH CONNECTION FOR USER: None
<myserver.example.com> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/jkirk/.ansible/cp/1e16484e09 -tt myserver.example.com 'apk --version'
<myserver.example.com> (255, '', 'Permission denied (publickey,password).\r\n')
fatal: [myserver.example.com]: UNREACHABLE! => {
    "changed": false, 
    "msg": "Failed to connect to the host via ssh: Permission denied (publickey,password).", 
    "unreachable": true
}
        to retry, use: --limit @/home/jkirk/projects/solsystem-ansible/bootstrap.retry
```

This used to work in version 2.11.0.

**TL;DR**: My current workaround is to set `bootstrap_remote_user: root` but the bootstrap role should also work with `-u root` option.

**Comment by jkirk** [comment]
@gregorydlogan There was an error in my previous comment:
```yaml
  roles:
    - robertdebock.bootstrap
      bootstrap_remote_user: root
```
needs to be:
```yaml
  roles:
    - role: robertdebock.bootstrap
      bootstrap_remote_user: root
```
I've edited my comment.

**Comment by jkirk** [comment]
I've just tested the lasted version 3.4.4 without `bootstrap_remote_user: root`, so running `ansible-playbook -u root bootstrap.yml` works for me again. Not sure about @gregorydlogan initial problem though.

**Comment by robertdebock** [comment]
Seems to be fixed.
